### PR TITLE
ACMS-1861: Fix Drupal Check notices for Drupal 10.1

### PR DESCRIPTION
**Pull request by rajeshreeputra**
**Motivation**
Fixes #ACMS-1861

**Proposed changes**
Fix Drupal Check notices for Drupal 10.1

**Alternatives considered**
NA

**Testing steps**
Check CI(drupal-check), it should pass.

**Merge requirements**
- [ ] _Major change_, _Minor change_, _Bug_, _Enhancement_, and/or _Chore_ label applied
- [ ] Manual testing by a reviewer

### CODE REVIEW

File: modules/acquia_cms_common/src/Services/ConfigImporterService.php [hunk 1]
```diff
@@ -209,7 +210,7 @@ public function doImport(StorageComparer $storage_comparer, DrushLoggerManager $
         $message = 'The import failed due to the following reasons:' . "\n";
         $message .= implode("\n", $config_importer->getErrors());
 
-        watchdog_exception('config_import', $e);
+        Error::logException('config_import', $e);
```
Comment (deepakmishra2) [inline modules/acquia_cms_common/src/Services/ConfigImporterService.php:213]: use DI here @rajeshreeputra

Comment (deepakmishra2) [reply modules/acquia_cms_common/src/Services/ConfigImporterService.php:213]: as per discussion we cannot use DI here because of static method implementation in Error class similar to Crypt class

File: tests/src/ExistingSite/InstallStateTest.php [hunk 2]
```diff
@@ -97,8 +97,8 @@ public function testConfig(): void {
     // Check page caching set for one year and css/js are aggregated.
     $performance_config = $this->config('system.performance');
     $this->assertSame(31536000, $performance_config->get('cache.page.max_age'));
-    $this->assertSame(TRUE, $performance_config->get('css.preprocess'));
-    $this->assertSame(TRUE, $performance_config->get('js.preprocess'));
+    $this->assertTrue(TRUE, $performance_config->get('css.preprocess'));
+    $this->assertTrue(TRUE, $performance_config->get('js.preprocess'));
```
Comment (chandan-singh7929) [inline tests/src/ExistingSite/InstallStateTest.php:101]: `assertTrue` does expect two condition, please use `$this->assertTrue($performance_config->get('css.preprocess'));` instead.

**Review by chandan-singh7929 — CHANGES_REQUESTED** [review]
Overall changes looks good to me, there are some minor comment please check.

File: tests/src/ExistingSite/InstallStateTest.php (hunk 2 above)
Comment (rajeshreeputra) [reply tests/src/ExistingSite/InstallStateTest.php]: done

**Review by apathak18 — COMMENTED** [review]
@rajeshreeputra Can we target only on the drupal-check fixes? PHPCS fixes could be done later.

cc: @chandan-singh7929 @vishalkhode1

File: modules/acquia_cms_common/src/Commands/AcmsCommands.php [hunk 3]
```diff
@@ -286,6 +286,7 @@ function ($key, $cellData, FormatterOptions $options, $rowData) {
       return $result;
     }
     $this->output()->writeln('Starter-kit is not installed yet.');
+    return FALSE;
```
Comment (apathak18) [inline modules/acquia_cms_common/src/Commands/AcmsCommands.php:289]: Revert this change. Not required

Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/Commands/AcmsCommands.php:289]: It is required as this method returns data, other it will throw "return statement is missing" error in drupal-check.

File: composer.json [hunk 4]
```diff
@@ -173,6 +170,9 @@
             },
             "mglaman/drupal-check": {
                 "Add custom rules to bypass check for buildForm": "https://github.com/mglaman/drupal-check/pull/286.patch"
+            },
+            "stecman/symfony-console-completion": {
+                "Return type fix for stecman/symfony-console-completion": "https://github.com/stecman/symfony-console-completion/pull/94.patch"
```
Comment (apathak18) [inline composer.json:175]: Revert this change, Not required ~~ Merged into starterkit

File: modules/acquia_cms_common/composer.json [hunk 5]
```diff
@@ -13,7 +13,7 @@
         "drupal/autologout": "^1.4",
         "drupal/config_ignore": "^3.0@beta",
         "drupal/config_rewrite": "^1.5",
-        "drupal/core": "^10.0.9",
+        "drupal/core": "^10.1",
```
Comment (apathak18) [inline modules/acquia_cms_common/composer.json:16]: @rajeshreeputra Not sure about this change I guess "^10" would work.

File: modules/acquia_cms_common/src/Commands/AcmsCommands.php (hunk 3 above)
Comment (apathak18) [reply modules/acquia_cms_common/src/Commands/AcmsCommands.php:289]: Already pushed into develop branch if you revert this change and check then it won't throw any error, also remove BOOL from return type of starterkit method

File: modules/acquia_cms_common/src/Services/ConfigImporterService.php (hunk 1 above)
Comment (apathak18) [inline modules/acquia_cms_common/src/Services/ConfigImporterService.php:213]: @rajeshreeputra prior to D10.1 this would throw an undefined method error ~~ can we add version_compare logic?
cc: @vishalkhode1

Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/Services/ConfigImporterService.php]: added version compare

File: composer.json (hunk 4 above)
Comment (rajeshreeputra) [reply composer.json:175]: it starterkit new version get released we need this.

File: modules/acquia_cms_common/composer.json (hunk 5 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_common/composer.json:16]: As discussed in DSM, minimum requirement is need to add as ^10.1

File: modules/acquia_cms_common/src/Services/ConfigImporterService.php [hunk 6]
```diff
@@ -15,6 +15,7 @@
 use Drupal\Core\Extension\ThemeHandlerInterface;
 use Drupal\Core\Lock\LockBackendInterface;
 use Drupal\Core\StringTranslation\TranslationInterface;
+use Drupal\Core\Utility\Error;
```
Comment (chandan-singh7929) [inline modules/acquia_cms_common/src/Services/ConfigImporterService.php:18]: Does this class available in below D10.1 version?

**Review by apathak18 — APPROVED** [review]

File: modules/acquia_cms_common/src/Commands/AcmsCommands.php (hunk 3 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/Commands/AcmsCommands.php]: removed.

File: modules/acquia_cms_common/src/Services/ConfigImporterService.php (hunk 6 above)
Comment (rajeshreeputra) [reply modules/acquia_cms_common/src/Services/ConfigImporterService.php:18]: yes it is available in 10.0

**Review by chandan-singh7929 — APPROVED** [review]

**Review by vishalkhode1 — APPROVED** [review]
